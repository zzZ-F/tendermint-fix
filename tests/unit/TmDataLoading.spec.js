import { mount } from "@vue/test-utils"
import htmlBeautify from "html-beautify"
import { TmDataLoading } from "@/index"

describe("TmDataLoading", () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(TmDataLoading)
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })

  it("has an icon", () => {
    expect(
      wrapper
        .find(".tm-data-msg__icon i.material-icons")
        .text()
        .trim()
    ).toBe("rotate_right")
  })

  it("has a spinning icon", () => {
    expect(
      wrapper.find(".tm-data-msg__icon").contains("i.material-icons.fa-spin")
    ).toBe(true)
  })

  it("has a title", () => {
    expect(
      wrapper
        .find(".tm-data-msg__title div")
        .text()
        .trim()
    ).toBe("Data is loadingâ€¦")
  })

  it("has a subtitle", () => {
    expect(
      wrapper
        .find(".tm-data-msg__subtitle div")
        .text()
        .trim()
    ).toBe("Please wait a moment.")
  })
})
