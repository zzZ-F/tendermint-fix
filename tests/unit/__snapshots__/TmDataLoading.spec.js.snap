// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmDataLoading has the expected html structure 1`] = `
<div class="tm-data-msg">
  <div class="tm-data-msg__icon">
    <i class="material-icons fa-spin">rotate_right</i>
  </div>
  <div class="tm-data-msg__text">
    <div class="tm-data-msg__title">
      <div>
        Data is loadingâ€¦
      </div>
    </div>
    <div class="tm-data-msg__subtitle">
      <div>
        Please wait a moment.
      </div>
    </div>
  </div>
</div>
`;
